pkg/channels/doc.go

```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+/*
+Package channels provides common utilities related to channels.
```

Comment:
Little bit hesitant to introduce a whole new package just for this. If anyone knows a place where this would fit let me know.